src/azure-cli/azure/cli/command_modules/search/custom.py

```diff
@@ -52,7 +54,12 @@ def create_search_service(cmd, resource_group_name, search_service_name, sku, lo
         _search.partition_count = partition_count
     if (public_network_access.lower() not in ["enabled", "disabled"]):
         raise ValueError("SearchService.PublicNetworkAccess: only [""enabled"", ""disabled""] are allowed")
+    if (hosting_mode not in ["default", "highDensity"]):
+        raise ValueError("SearchService.HostingMode: only [""default"", ""highDensity""] are allowed")
```

Comment:
Please use error types defined in this [guideline](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md#available-error-types)